### DISCUSSION TITLE
Push based continuous queries

### DISCUSSION THREAD

**Discussion by sirinath** [databendlabs/databend#8504] [Ideas]
Generally database queries and pull based where the database would poll to see any updates. For ultra low latency high throughput real time applications this is in efficient. Ideally we should have push based continuous queries. Here when data changes the changes are notified to application.

**Comment by jovany-wang** [comment]
Looks interesting! It'd be great if you can share more docs or more details to help us understand it. :)

**Comment by sirinath** [comment]
This concept is taken from https://rethinkdb.com/. It is descried in https://rethinkdb.com/blog/realtime-web/.

**Comment by sundy-li** [comment]
Sounds like noria, https://www.youtube.com/watch?v=s19G6n0UjsM&t=1934s

**Comment by sirinath** [comment]
Yes. [Noria](https://github.com/mit-pdos/noria) looks similar.

**Comment by ChenpiDog** [comment]
Real-time Data Warehouse and real-time ETL can be realized only with SQL. Some commercial data warehouse products have launched this function, such as Alibaba Cloud AnalyticDB PostgreSQL and Alibaba Cloud Hologres. You can refer to the real-time materialized views of these two products.
Alibaba Cloud AnalyticDB PostgreSQL: https://help.aliyun.com/document_detail/253369.html

**Comment by ChenpiDog** [comment]
![Databend沟通 drawio](https://user-images.githubusercontent.com/55483982/193726108-86564cc8-9915-428d-a5e0-dc9971ee8461.png)
